src/plugins/fs.c

```diff
@@ -2966,3 +3017,217 @@ gboolean bd_fs_vfat_resize (const gchar *device, guint64 new_size, GError **erro
+ */
+gboolean bd_fs_ntfs_resize (const gchar *device, guint64 new_size, GError **error) {
+    const gchar *args[5] = {"ntfsresize", NULL, NULL, NULL, NULL};
+    gboolean ret = FALSE;
+
+    if (!check_deps (&avail_deps, DEPS_NTFSRESIZE_MASK, deps, DEPS_LAST, &deps_check_lock, error))
+        return FALSE;
+
+    if (new_size != 0) {
+        args[1] = "-s";
+        args[2] = g_strdup_printf ("%"G_GUINT64_FORMAT, new_size);
+        args[3] = device;
+    } else {
+        args[1] = device;
+    }
+    ret = bd_utils_exec_and_report_error (args, NULL, error);
+
+    g_free ((gchar *) args[2]);
+    return ret;
+}
+
+/**
+ * bd_fs_ntfs_get_info:
+ * @device: the device containing the file system to get info for (device must
+            not be mounted, trying to get info for a mounted device will result
+            in an error)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: (transfer full): information about the file system on @device or
+ *                           %NULL in case of error
+ *
+ * Tech category: %BD_FS_TECH_NTFS-%BD_FS_TECH_MODE_QUERY
+ */
+BDFSNtfsInfo* bd_fs_ntfs_get_info (const gchar *device, GError **error) {
+    const gchar *args[3] = {"ntfscluster", device, NULL};
+    gboolean success = FALSE;
+    gchar *output = NULL;
+    BDFSNtfsInfo *ret = NULL;
+    gchar **lines = NULL;
+    gchar **line_p = NULL;
+    gchar *val_start = NULL;
+    g_autofree gchar* mountpoint = NULL;
+
+    if (!check_deps (&avail_deps, DEPS_NTFSCLUSTER_MASK, deps, DEPS_LAST, &deps_check_lock, error))
+        return FALSE;
+
+    mountpoint = bd_fs_get_mountpoint (device, error);
+    if (mountpoint != NULL) {
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_NOT_MOUNTED,
+                     "Can't get NTFS file system information for '%s': Device is mounted.", device);
+        return NULL;
+    } else {
+        if (*error != NULL) {
+            g_prefix_error (error, "Error when trying to get mountpoint for '%s': ", device);
+            return NULL;
+        }
+    }
+
+    success = bd_utils_exec_and_capture_output (args, NULL, &output, error);
+    if (!success)
+        /* error is already populated */
+        return FALSE;
+
+    ret = g_new0 (BDFSNtfsInfo, 1);
+    lines = g_strsplit (output, "\n", 0);
+    g_free (output);
+    line_p = lines;
+    /* find the beginning of the (data) section we are interested in */
+    while (*line_p && !g_str_has_prefix (*line_p, "bytes per volume"))
+        line_p++;
+    if (!line_p) {
+        /* error is already populated */
+        g_set_error (error, BD_FS_ERROR, BD_FS_ERROR_PARSE, "Failed to parse NTFS file system information");
+        g_strfreev (lines);
+        bd_fs_ntfs_info_free (ret);
+        return FALSE;
+    }
+
+    /* extract data from something like this: "bytes per volume        : 998240256" */
+    val_start = strchr (*line_p, ':');
+    val_start++;
+    while (isspace (*val_start))
+        val_start++;
```

Comment:
I think this is not necessary, spaces before the number should be fine. But please verify if that's really the case. :)